frontend/src/pages/signup/Signup.tsx

```diff
@@ -1,54 +1,87 @@
+import EmailIcon from '@mui/icons-material/Email';
+
 import {
   ContentHeader,
   ContentBox,
   Lines,
   FormTextField,
 } from 'src/components';
+import NotificationsEmailResearch from 'src/features/settings/preferences/fields/NotificationsEmailResearch';
+import NotificationsEmailNewFeatures from 'src/features/settings/preferences/fields/NotificationsEmailNewFeatures';
+import NotificationsLang from 'src/features/settings/preferences/fields/NotificationsLang';
 import { useNotifications } from 'src/hooks';
-import {
-  AccountsService,
-  ApiError,
-  RegisterUserRequest,
-} from 'src/services/openapi';
-import { Link } from 'react-router-dom';
-
+import { AccountsService, ApiError } from 'src/services/openapi';
 import { TRACKED_EVENTS, trackEvent } from 'src/utils/analytics';
+import { defaultNotificationsLanguage } from 'src/utils/userSettings';
 
 const SignupSuccess = ({ email }: { email: string }) => {
   const { t } = useTranslation();
   return (
-    <Typography>
+    <Alert severity="success">
+      <AlertTitle>{t('signup.welcome')}</AlertTitle>
       <Trans t={t} i18nKey="signup.successMessage">
-        Success!
-        <br />A verification link has been sent to <code>{{ email }}</code> .
+        A verification link has been sent to <code>{{ email }}</code> .
       </Trans>
-    </Typography>
+    </Alert>
   );
 };
 
 const Signup = () => {
   const { t } = useTranslation();
+  const { displayErrorsFrom } = useNotifications();
 
   const [apiError, setApiError] = useState<ApiError | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
   const [successEmailAddress, setSuccessEmailAddress] = useState<string | null>(
     null
   );
+
   const [acceptPolicy, setAcceptPolicy] = useState(false);
-  const [isLoading, setIsLoading] = useState(false);
-  const { displayErrorsFrom } = useNotifications();
+
+  // Legally, the notification settings must be false by default.
+  const [notififLang, setNotififLang] = useState(
+    defaultNotificationsLanguage()
+  );
+  const [notififResearch, setNotififResearch] = useState(false);
+  const [notifNewFeatures, setNnotifNewFeatures] = useState(false);
 
   const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     setApiError(null);
     setIsLoading(true);
+
     const formData = new FormData(event.currentTarget);
-    const formObject: unknown = Object.fromEntries(formData);
+
     try {
       const createdUser = await AccountsService.accountsRegisterCreate({
-        requestBody: formObject as RegisterUserRequest,
+        requestBody: {
+          email: (formData.get('email') as string) ?? '',
+          username: (formData.get('username') as string) ?? '',
+          password: (formData.get('password') as string) ?? '',
+          password_confirm: (formData.get('password_confirm') as string) ?? '',
+          settings: {
+            general: {
+              notifications__lang: notififLang,
```

Comment:
As discussed I would use the current UI language to initialize this settings (i.e use `i18n.resolvedLanguage`).

It would also make unnecessary to add one more language detection mechanism in `isNavigatorLang()`.